Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -432,17 +439,30 @@ WebIDL::ExceptionOr<void> XMLHttpRequest::send(Optional<Fetch::XMLHttpRequestBod
 
     auto request = LoadRequest::create_for_url_on_page(request_url, m_window->page());
     request.set_method(m_method);
-    if (body_with_type.has_value()) {
-        TRY_OR_RETURN_OOM(realm, body_with_type->body.source().visit([&](ByteBuffer const& buffer) -> ErrorOr<void> {
-                request.set_body(buffer);
-                return {}; }, [&](JS::Handle<FileAPI::Blob> const& blob) -> ErrorOr<void> {
-                auto byte_buffer = TRY(ByteBuffer::copy(blob->bytes()));
-                request.set_body(byte_buffer);
-                return {}; }, [](auto&) -> ErrorOr<void> { return {}; }));
-        if (body_with_type->type.has_value()) {
-            // If type is non-null and this’s headers’s header list does not contain `Content-Type`, then append (`Content-Type`, type) to this’s headers.
-            if (!m_request_headers.contains("Content-Type"sv))
-                request.set_header("Content-Type", String { body_with_type->type->span() });
+    if (serialized_document.has_value()) {
+        request.set_body(move(serialized_document.value()));
+    } else if (body_with_type.has_value()) {
+        TRY(body_with_type->body.source().visit(
+            [&](ByteBuffer const& buffer) -> WebIDL::ExceptionOr<void> {
+                request.set_body(move(buffer));
```

Comment:
This still ends up copying, you can't `move()` a reference like that. The copy ctor really needs to go.

```suggestion
                auto byte_buffer = TRY_OR_RETURN_OOM(realm, ByteBuffer::copy(buffer));
                request.set_body(move(byte_buffer));
```